explorer/interpreter/interpreter.cpp

```diff
@@ -84,7 +84,7 @@ class Interpreter {
   // State transitions for expressions.
   auto StepExp() -> ErrorOr<Success>;
   // State transitions for lvalues.
-  auto StepLvalue() -> ErrorOr<Success>;
+  auto StepLocation() -> ErrorOr<Success>;
```

Comment:
Seems fine for now, though I think we'll probably eventually want these to be `StepValueExp` and `StepReferenceExp`, with a third function added for `StepInitializingExp`.